juju/juju
# User Feedback for advance-generation Invoking Generation Completion

## Description of change

- When an operator uses the `advance-generation` command and unit advancement causes auto-completion to be invoked, this result is reported via stdout.
- If the generation is successfully completed, the active generation is set as "current" in the local store.

Depends on https://github.com/juju/juju/pull/9725.

## QA steps

- `export JUJU_DEV_FEATURE_FLAGS=generations`.
- Bootstrap.
- `juju deploy redis -n 2`
- `juju add-generation` will set the target generation to "next".
- `juju advance-generation redis/0` will result in no extra stdout feedback.
- `juju advance-generation redis` (all remaining will auto-complete generation).
- Ensure that the completion notification is written to stdout.
- Ensure that ~/.local/share/juju/models shows generation "current".

## Documentation changes

None.

## Bug reference

N/A
